src/_nebari/stages/infrastructure/__init__.py

```diff
@@ -543,6 +544,7 @@ class AWSNodeGroup(NodeGroup):
     gpu: bool = False
     single_subnet: bool = False
     permissions_boundary: Optional[str] = None
+    capacity_type: Optional[str] = None
```

Comment:
The only thing I would change in here, specifically since this is user-facing (validation), would be to change 
```suggestion
    capacity_type: Optional[Literal['SPOT', 'ON_DEMAND']] = None
```
This ensures that if a user makes an incorrect typo or misplacement in the config, we fail during the Python step instead of the Terraform part.